favorites/models.py

```diff
@@ -0,0 +1,21 @@
+from django.db import models
+from django.contrib.auth.models import User
+from core.models import BaseModel
+from package.models import Package
+
+
+class Favorite(BaseModel):
+    favorited_by = models.ForeignKey(
+        User, blank=True, null=True, on_delete=models.SET_NULL
+    )
+    package = models.ForeignKey(Package, on_delete=models.CASCADE)
+
+    def save(self, *args, **kwargs):
+        super().save(*args, **kwargs)
+        self.package.favorite_count += 1
```

Comment:
Not sure if you have worked with `F()` expressions before. https://docs.djangoproject.com/en/5.0/ref/models/expressions/#f-expressions

They are really nice for changing relative values because Django will create an SQL statement that modifies the value that is stored in the database table at the time the expression is run versus whatever values are stored in the package.favorite_count at the time of the change.